File: 2022/javascript/day01/solve.js
const fs = require("fs");
const input = fs.readFileSync("input.txt", { encoding: "utf8" });

const invs = input
	.split("\n\n")
	.map((inv) => inv.split("\n").reduce((sum, line) => sum + parseInt(line), 0))
	.sort((a, b) => b - a);

console.log("Part 1:", invs[0]);
console.log("Part 2:", invs[0] + invs[1] + invs[2]);

File: 2022/go/day01/solve.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	content, _ := os.ReadFile("input.txt")
	invs := make([]int, 0)

	for _, inv := range strings.Split(string(content), "\n\n") {
		items := strings.Split(inv, "\n")
		totalCals := 0
		for _, item := range items {
			cals, _ := strconv.Atoi(item)
			totalCals += cals
		}

		invs = append(invs, totalCals)

	}

	sort.Slice(invs, func(i, j int) bool {
		return invs[i] > invs[j]
	})

	fmt.Println("Part 1:", invs[0])
	fmt.Println("Part 2:", invs[0]+invs[1]+invs[2])
}
